elements/pf-back-to-top/test/pf-back-to-top.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { expect, html, nextFrame, fixture } from '@open-wc/testing';
+import { createFixture } from '@patternfly/pfe-tools/test/create-fixture.js';
+import { setViewport } from '@web/test-runner-commands';
+
+import { allUpdates } from '@patternfly/pfe-tools/test/utils.js';
+
+import { PfBackToTop } from '../pf-back-to-top.js';
+
+describe('<pf-back-to-top>', function() {
+  let element: PfBackToTop;
+  let link: HTMLAnchorElement;
+
+  beforeEach(async function() {
+    await setViewport({ width: 320, height: 640 });
+    window.scrollTo({ top: 0, behavior: 'instant' });
+    await nextFrame();
+  });
+
+  describe('default a tag implementation', function() {
+    beforeEach(async function() {
+      await fixture<HTMLElement>(html`<div style="height: 2000px;"></div>`);
+      element = await createFixture<PfBackToTop>(html`<pf-back-to-top href="#top">Back to Top</pf-back-to-top>`);
+      await allUpdates(element);
+      link = element.shadowRoot!.querySelector('a')!;
+    });
+
+    it('imperatively instantiates', function() {
+      expect(document.createElement('pf-back-to-top')).to.be.an.instanceof(PfBackToTop);
+    });
+
+    it('should upgrade', async function() {
+      const klass = customElements.get('pf-back-to-top');
+      expect(element)
+        .to.be.an.instanceOf(klass)
+        .and
+        .to.be.an.instanceOf(PfBackToTop);
+    });
+
+    it('should be visible after 400px scroll', async function() {
```

Comment:
describe/beforeEach to setup state, it/expect to examine state. use nested describes.